tuf/data/interface.go

```diff
@@ -0,0 +1,28 @@
+package data
+
+// TufRole is an interface for dealing with TUF roles internally
+type TufRole interface {
+	// Returns true for BaseRole
+	IsBaseRole() bool
```

Comment:
This seems weird. Shouldn't we simply rely on casting to verify if a Role is a of a specific type? Seems like the go way, and would remove `IsBaseRole()` and `IsDelegationRole()` here and in BaseRole and DelegationRole.